### PR TITLE
Reconstruct converter

### PR DESCRIPTION
**Pull request by zhber**
reconstruct converter module in Sparsebit.

- add `InputMatchType.SUBSET` support
- add node with multiple output branches support
- more modular code
- some changes in API names

### CODE REVIEW

File: sparsebit/quantization/converters/fuse_operations/disable_nearest_upsapmle_quant.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import torch
 
-from ..base import ReplacePatternBase, MatcherNode
+from ..utils import ReplacePatternBase, MatchingNode
```
Comment (PeiqinSun) [inline sparsebit/quantization/converters/fuse_operations/disable_nearest_upsapmle_quant.py:3]: please update absolute path

File: sparsebit/quantization/converters/utils/disjoint_set_union.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        self.n = nums
+        assert (
+            isinstance(value, list) and len(value) == nums or value is None
+        ), "Unknown format"
+        self.with_value = value is not None
+        if self.with_value:
+            self.value = value
+            assert cmp is not None, "Unknown format"
+            self.cmp = cmp
+        self.fa = [i for i in range(self.n)]
```
Comment (PeiqinSun) [inline sparsebit/quantization/converters/utils/disjoint_set_union.py:20]: parents

File: sparsebit/quantization/converters/utils/subgraph_matching_utils.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+    Args:
+        repeat (bool): 是否要循环调用到网络中不再出现子图为止。
+    """
+
+    def __init__(self, repeat: bool):
+        self.repeat = repeat
+
+
+class ReplaceStrategy(object):
```
Comment (PeiqinSun) [inline sparsebit/quantization/converters/utils/subgraph_matching_utils.py:29]: please use enum

File: sparsebit/quantization/converters/utils/subgraph_matching_replace_pattern.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+            nodes = list(m.graph.nodes)
+            replace_dict = self.get_new_graph(
+                nodes_dict=nodes_dict,
+                modules_dict=modules_dict,
+                model=m,
+                transform_idx=self.timer.get_idx(),
+            )
+            for rep_name, new_node in replace_dict.items():
+                replace_op(nodes, nodes_dict[rep_name], new_node)
+            is_transformed = True
```
Comment (PeiqinSun) [inline sparsebit/quantization/converters/utils/subgraph_matching_replace_pattern.py:111]: why reset is_transformed?

File: sparsebit/quantization/converters/utils/dominator_tree.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        # edges (idom[x], x) is a tree, except for (idom[n-1]=n-1, n-1) is loop
+        self.finish_flag = True
+        self.rebuild()
+
+    def rebuild(self):
+        assert self.finish_flag
+        sons = [0] * self.n
+        masks = [0] * self.n
+
+        def dfs2(x: int):
```
Comment (PeiqinSun) [inline sparsebit/quantization/converters/utils/dominator_tree.py:83]: dfs2 -> ?

File: sparsebit/quantization/converters/utils/dominator_tree.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+            sons[x] += 1
+            for i in self.targets[x]:
+                dfs2(i)
+                sons[x] += sons[i]
+
+        def dfs3(x: int):
```
Comment (PeiqinSun) [inline sparsebit/quantization/converters/utils/dominator_tree.py:89]: dfs3 -> ?

File: sparsebit/quantization/converters/fuse_operations/disable_nearest_upsapmle_quant.py (hunk 1 above)
Comment (zhber) [reply sparsebit/quantization/converters/fuse_operations/disable_nearest_upsapmle_quant.py]: fixed

File: sparsebit/quantization/converters/utils/disjoint_set_union.py (hunk 2 above)
Comment (zhber) [reply sparsebit/quantization/converters/utils/disjoint_set_union.py]: fixed

File: sparsebit/quantization/converters/utils/subgraph_matching_replace_pattern.py (hunk 4 above)
Comment (zhber) [reply sparsebit/quantization/converters/utils/subgraph_matching_replace_pattern.py]: removed

File: sparsebit/quantization/converters/utils/dominator_tree.py (hunk 5 above)
Comment (zhber) [reply sparsebit/quantization/converters/utils/dominator_tree.py]: fixed

File: sparsebit/quantization/converters/utils/dominator_tree.py (hunk 6 above)
Comment (zhber) [reply sparsebit/quantization/converters/utils/dominator_tree.py]: fixed

**Pull request merged**